src/bootstrap/lib.rs

```diff
@@ -368,7 +368,7 @@ impl Build {
         // https://github.com/rust-lang/rust/blob/a8a33cf27166d3eabaffc58ed3799e054af3b0c6/src/bootstrap/bootstrap.py#L796-L797
         let is_sudo = match env::var_os("SUDO_USER") {
             Some(_sudo_user) => {
-                let uid = unsafe { libc::getuid() };
+                let uid = nix::unistd::Uid::current().as_raw();
```

Comment:
It's literally doing the same thing with couple of abstractions on top of it. Adding a new dependency just to achieve the same result (even in a slower way with a longer call chain) doesn't seem right to me.

ref https://docs.rs/nix/latest/src/nix/unistd.rs.html#1442-1450

Reply:
I agree with you and I believe the comments on the issue itself stated the same thing, an oversight by me.